src/commands/cmd_key.cc

```diff
@@ -553,6 +553,68 @@ class CommandSort : public Commander {
   SortArgument sort_argument_;
 };
 
+class CommandGetMeta : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() != 2) {
+      return {Status::RedisExecErr, errWrongNumOfArguments};
+    }
+    return Status::OK();
+  }
+
+  Status Execute(engine::Context &ctx, Server *srv, Connection *conn, std::string *output) override {
+    redis::Database redis(srv->storage, conn->GetNamespace());
+    std::string &key = args_[1];
+    std::string nskey = redis.AppendNamespacePrefix(key);
+
+    RedisType type = kRedisNone;
+    auto s = redis.Type(ctx, key, &type);
+    if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
+    if (type == kRedisNone) {
+      *output = conn->NilString();
+      return Status::OK();
+    }
+
+    // Create appropriate metadata object based on type
+    std::unique_ptr<Metadata> metadata;
+    switch (type) {
+      case kRedisString:
+        metadata = std::make_unique<StringMetadata>();
+        break;
+      case kRedisHash:
+        metadata = std::make_unique<HashMetadata>();
+        break;
+      case kRedisSet:
+        metadata = std::make_unique<SetMetadata>();
+        break;
+      case kRedisZSet:
+        metadata = std::make_unique<ZSetMetadata>();
+        break;
+      case kRedisBitmap:
+        metadata = std::make_unique<BitmapMetadata>();
+        break;
+      case kRedisList:
+        metadata = std::make_unique<ListMetadata>();
+        break;
+      default:
+        return {Status::RedisExecErr, "Unimplemented Redis type"};
+    }
+
+    // Get metadata
+    s = redis.GetMetadata(ctx, {type}, nskey, metadata.get());
```

Comment:
Since you don't need other metadata fields, you can just use `GetMetadata(kRedisNone)` and remove the `redis.Type` call & the switch statement.

Reply:
for my current use case I'm working with redis hashes.
I added this other cases to make the cmd support future use cases that someone else (or myself) may have to work on.
is it a blocker to leave this is?

Reply:
i'm not following completely this suggestion do you mind giving more details?